plugins/services/src/js/components/MesosLogView.js

```diff
@@ -115,8 +115,8 @@ class MesosLogView extends mixin(StoreMixin) {
       (props.logName !== nextProps.logName) ||
       // Check task (slave_id is the only property being used)
       (props.task.slave_id !== nextProps.task.slave_id) ||
-      // Check hasLoadingError
-      (state.hasLoadingError !== nextState.hasLoadingError) ||
+      // Check mesosLogXHRError
+      (state.mesosLogXHRError !== nextState.mesosLogXHRError) ||
```

Comment:
No it will reference the same JS `Object`. The statement you make above is using 2 Object literals to create 2 new Objects, so yes your statement is correct. However, `state.mesosLogXHRError` and `nextState.mesosLogXHRError` will either point to the same object or not and that's the test we want.